pkg/messaging/rabbitmq/pubsub.go

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) Mainflux
+// SPDX-License-Identifier: Apache-2.0
+
+package rabbitmq
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/gogo/protobuf/proto"
+	log "github.com/mainflux/mainflux/logger"
+	"github.com/mainflux/mainflux/pkg/messaging"
+	amqp "github.com/rabbitmq/amqp091-go"
+)
+
+// SubjectAllChannels represents subject to subscribe for all the channels.
+const (
+	ChansPrefix        = "channels"
+	SubjectAllChannels = "channels.>"
+	RoutingKey         = "application specific routing key for fancy topologies"
+	Exchange           = "mainflux"
+	ExchangeKind       = "fanout"
+	QueueDurability    = true
+	QueueDelete        = false
+	QueueExclusivity   = false
+	QueueWait          = false
+	ConsumerTag        = "mainflux-consumer"
+	Mandatory          = false
+	Immediate          = false
```

Comment:
Is there any reason these constants are exported?

Reply:
No reason. I am changing that

Reply:
Done